test/subscription_map.cpp

```diff
@@ -0,0 +1,284 @@
+// Copyright Takatoshi Kondo 2020
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+#include "test_main.hpp"
+#include "combi_test.hpp"
+#include "checker.hpp"
+#include "global_fixture.hpp"
+
+#include "subscription_map.hpp"
+#include "retained_topic_map.hpp"
+
+#include <iostream>
+
+BOOST_AUTO_TEST_SUITE(test_subscription_map)
+
+BOOST_AUTO_TEST_CASE( failed_erase ) {
+    using func_t = std::function<void()>;
+    using value_t = std::shared_ptr<func_t>; // shared_ptr for '<' and hash
+    using sm_t = multiple_subscription_map<value_t>;
+
+    sm_t m;
+    auto v1 = std::make_shared<func_t>([] { std::cout << "v1" << std::endl; });
+    auto v2 = std::make_shared<func_t>([] { std::cout << "v2" << std::endl; });
+
+    BOOST_TEST(m.size() == 1);
+    auto it_success1 = m.insert("a/b/c", v1);
+    assert(it_success1.second);
+    BOOST_TEST(m.size() != 1);
+
+    auto it_success2 = m.insert("a/b", v2);
+    assert(it_success2.second);
+    BOOST_TEST(m.size() != 1);
+
+    auto e1 = m.erase(it_success1.first, v1);
+    BOOST_TEST(m.size() != 1);
+
+    auto e2 = m.erase(it_success2.first, v2); //  Invalid handle was specified is thrown here
+    BOOST_TEST(m.size() == 1);
+
+}	
+
+BOOST_AUTO_TEST_CASE( TestSingleSubscription ) {
```

Comment:
```suggestion
BOOST_AUTO_TEST_CASE( test_signle_subscription ) {
```
Could you use snake_case instead of CamenCase ?